source/scripts/main.js

```diff
@@ -0,0 +1,196 @@
+var $SOMenuVisibility = "hidden";
+
+function toggleMenu() {
+    var menuIcon = document.getElementById("menu-icon");
+    menuIcon.classList.toggle("change");
+
+    var slideOverMenu = document.getElementById("slide-over-menu");
+
+    if ($SOMenuVisibility == "hidden") {
+        slideOverMenu.style.transform = "translate(100%)";
+        $SOMenuVisibility = "visible";
+    }
+    else {
+        slideOverMenu.style.transform = "translate(-100%)";
+        $SOMenuVisibility = "hidden";
+    }
+}
+
+function showSettings() {
+    hideHome();
+    hideCookbooks();
+    hideRecipeCards();
+    // hideRecipePage();
+    const settings = document.getElementById("settings-container");
+    settings.style.visibility = "visible";
+    //settings.style.transform = "translate(100%)";
+
+    // Get the list of restrictions from local storage
+    const getDietaryRestrictions = JSON.parse(localStorage.getItem("dietaryRestrictions"));
+    const getIntolerancesRestrictions = JSON.parse(localStorage.getItem("intolerancesRestrictions"));
+    const dietaryContainerElements = document.getElementById("dietary-container").elements;
+    const intolerancesContainerElements = document.getElementById("intolerances-container").elements;
+
+    for (let i = 0; i < dietaryContainerElements.length; i++) {
+        const dietaryRestriction = dietaryContainerElements[i];
+        // If our restriction is in the list, then check it on the page
+        if (getDietaryRestrictions && getDietaryRestrictions.includes(dietaryRestriction.value)) {
+            dietaryRestriction.checked = true;
+        }
+    }
+
+    for (let i = 0; i < intolerancesContainerElements.length; i++) {
+        const intoleranceRestriction = intolerancesContainerElements[i];
+        // If our restriction is in the list, then check it on the page
+        if (getIntolerancesRestrictions && getIntolerancesRestrictions.includes(intoleranceRestriction.value)) {
+            intoleranceRestriction.checked = true;
+        }
+    }
+}
+
+function clearCheckBoxes() {
+    let checkboxes = document.querySelector("#recipe-page-container > recipe-page")
+        .shadowRoot.querySelectorAll("#ingredients-list > ul > ol > input");
+    checkboxes.forEach(e => e.checked = false);
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422335)
```suggestion
    checkboxes.forEach((e) => e.checked = false);
```